### DISCUSSION TITLE
How to add a nuget dependency?

### DISCUSSION THREAD

**Discussion by srid** [TheAngryByrd/MiniScaffold#236] [Q&A] [answered]
Here's what I tried,

1. Add `nuget Memstate.All` to `paket.dependencies`
2. Run `./build.sh DotnetRestore` (I ran this inside the devcontainer, FWIW)

But the added package cannot be referenced from the source. Perhaps I'm missing something? The [paket manual](https://fsprojects.github.io/Paket/learn-how-to-use-paket.html#Important-paket-commands) states that I should run `paket install` - but I'd expect the project build script (via build.fsx) to do it for me? Also, the `paket` command is not available in the docker devcontainer.

**Comment by TheAngryByrd** [comment, marked as answer]
### Short Answer:

I think the only step you need from your current process is to add it to [paket.references](https://fsprojects.github.io/Paket/references-files.html) as well. So add ` Memstate.All` all paket.references files in your projs that require it.

You can also do `dotnet paket add Memstate.All -i`, this will add this package and allow you to choose which projects should this dependency be added to.



### Longer Answer

To add a nuget package via paket, you'll want to use [paket add](https://fsprojects.github.io/Paket/paket-add.html). You can either specify `-i` flag to interactively choose with projects to add this to or `--project <path-to-project>` to choose a specific project.  

Paket is installed as a [dotnet tool](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-install) so you'll want to prefix any paket command with `dotnet`. If you get an error saying `dotnet paket` doesn't exist you'll need to run `dotnet tool restore` (build.cmd/sh does this for you) to make sure you have the dotnet tools. If there's still a bug with the docker devcontainer after this please open an issue).
